webpack/assets/javascripts/react_app/API.js

```diff
@@ -0,0 +1,12 @@
+import ServerActions from './actions/ServerActions';
+
+export default {
+  getStatisticsData(url) {
+    $.getJSON(url)
+      .success(
+        rawStatistics => ServerActions.receivedStatistics(rawStatistics)
+      )
+      // eslint-disable-next-line no-console
+      .error(error => console.log(error));
```

Comment:
I would suggest that on error you still trigger the ServerActions call, but with the error, maybe you could return a data structure that support both data an error message, and when success its error is empty, and when on failure the data is empty?